src/librustc_mir/build/expr/stmt.rs

```diff
@@ -161,6 +162,41 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                 );
                 block.unit()
             }
+            ExprKind::PlaceTypeAscription { source, user_ty } => {
+                let place = unpack!(block = this.as_place(block, source));
+                this.cfg.push(
+                    block,
+                    Statement {
+                        source_info,
+                        kind: StatementKind::AscribeUserType(
+                            place,
+                            Variance::Invariant,
+                            user_ty,
+                        ),
+                    },
+                );
+                block.unit()
+            }
+            ExprKind::ValueTypeAscription { source, user_ty } => {
+                let source = this.hir.mirror(source);
+                let source_span = source.span;
+                let source_ty = source.ty.clone();
+                let temp = this.temp(source_ty.clone(), source_span);
+                unpack!(block = this.into(&temp, block, source));
```

Comment:
Here I think you would call `as_temp` instead of calling `into` yourself:

https://github.com/rust-lang/rust/blob/1002e404e1ae6b8b907c8655edd41380d0c561cb/src/librustc_mir/build/expr/as_temp.rs#L19-L27